src/shell/server.cpp

```diff
@@ -500,9 +491,10 @@ class server::info_task : public task<unit> {
                 auto opts = m_server->m_ios.get_options();
                 auto env = m_server->m_initial_env;
                 if (auto mod = m_mod_info->m_result.peek()) {
-                    if (mod->m_loaded_module->m_env)
-                        env = mod->m_loaded_module->m_env.get();
-                    if (!mod->m_snapshots.empty()) opts = mod->m_snapshots.back()->m_options;
+                    if (!mod->m_snapshots.empty()) {
```

Comment:
Right, in that case we should fall back to the completed environment if available.